frontend/src/app/main/ui/static.cljs

```diff
@@ -10,49 +10,273 @@
    [app.common.data :as d]
    [app.common.pprint :as pp]
    [app.common.uri :as u]
+   [app.main.data.common :as dc]
    [app.main.data.events :as ev]
    [app.main.store :as st]
+   [app.main.ui.auth.login :refer [login-methods]]
+   [app.main.ui.auth.recovery-request :refer [recovery-request-page recovery-sent-page]]
+   [app.main.ui.auth.register :refer [register-methods register-validate-form register-success-page terms-register]]
+   [app.main.ui.dashboard.sidebar :refer [sidebar]]
    [app.main.ui.icons :as i]
+   [app.main.ui.viewer.header :as header]
    [app.util.dom :as dom]
    [app.util.globals :as globals]
    [app.util.i18n :refer [tr]]
    [app.util.router :as rt]
    [app.util.webapi :as wapi]
+   [cuerdas.core :as str]
    [potok.v2.core :as ptk]
    [rumext.v2 :as mf]))
 
 (mf/defc error-container
   {::mf/wrap-props false}
   [{:keys [children]}]
-  (let [on-click (mf/use-callback #(set! (.-href globals/location) "/"))]
+  (let [profile-id (:profile-id @st/state)
+        on-click (mf/use-fn #(set! (.-href globals/location) "/"))]
     [:section {:class (stl/css :exception-layout)}
      [:button
       {:class (stl/css :exception-header)
        :on-click on-click}
-      i/logo-icon]
+      i/logo-icon
+      (when profile-id
+        (str "< "
+             (tr "not-found.no-permission.go-dashboard")))]
      [:div {:class (stl/css :deco-before)} i/logo-error-screen]
+     (when-not profile-id
+       [:button {:class (stl/css :login-header)
+                 :on-click on-click}
+        (tr "labels.login")])
 
      [:div {:class (stl/css :exception-content)}
       [:div {:class (stl/css :container)} children]]
 
-     [:div {:class (stl/css :deco-after)} i/logo-error-screen]]))
+     [:div {:class (stl/css :deco-after2)}
+      [:span (tr "not-found.copyright")]
+      i/logo-error-screen
+      [:span (tr "not-found.made-with-love")]]]))
 
 (mf/defc invalid-token
   []
   [:> error-container {}
    [:div {:class (stl/css :main-message)} (tr "errors.invite-invalid")]
    [:div {:class (stl/css :desc-message)} (tr "errors.invite-invalid.info")]])
 
+
+
+(mf/defc login-dialog
+  [{:keys [show-dialog]}]
+  (let [current-section  (mf/use-state :login)
+        user-email (mf/use-state "")
+        register-token (mf/use-state "")
+        on-close
+        (mf/use-fn
+         #(set! (.-href globals/location) "/"))
```

Comment:
This function is used in many times, I guess it is better to make it a standalone funcion on app.util.router ?